frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -3717,21 +3752,31 @@ static bool resolveFnCallSpecial(Context* context,
     }
   }
 
-  if ((ci.name() == USTR("==") || ci.name() == USTR("!=")) &&
-      ci.numActuals() == 2) {
-    auto lhs = ci.actual(0).type();
-    auto rhs = ci.actual(1).type();
+  if ((ci.name() == USTR("==") || ci.name() == USTR("!="))) {
+    if (ci.numActuals() == 2) {
+      auto lhs = ci.actual(0).type();
+      auto rhs = ci.actual(1).type();
```

Comment:
Why not do something like this:

```C++
// foo.type == ?
CHPL_ASSERT(ci.numActuals() == 1 || ci.numActuals() == 2);
auto lhs = ci.actual(0).type();
auto rhs = ci.hasQuestionArg() ? QualifiedType(QualifiedType::TYPE, AnyType::get(context)) : 
                                 ci.actual(1).type();
```

And then the rest of the logic is the same along both paths? I know it's technically a bit sower but surely it won't matter too much.